### PR TITLE
Change background blur and replacement behaviors to clone parameter object

### PR DESCRIPTION
**Pull request by hekmatba**
**Issue #:**

**Description of changes:**
Change background blur and replacement `create` and `isSupported` interfaces to clone spec and option parameters instead of changing the objects by reference. 

**Testing:**

*Can these tested using a demo application? Please provide reproducible step-by-step instructions.*
Verify blur and replacement are working.

**Checklist:**

1. Have you successfully run `npm run build:release` locally?
yes

2. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved?
no

3. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved?
no

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/backgroundblurprocessor/BackgroundBlurVideoFrameProcessor.ts [hunk 1]
```diff
@@ -62,11 +62,17 @@ export default class BackgroundBlurVideoFrameProcessor extends BackgroundFilterV
     spec?: BackgroundFilterSpec,
     options?: BackgroundBlurOptions
   ): Promise<BackgroundBlurProcessor | undefined> {
-    spec = BackgroundBlurVideoFrameProcessor.resolveSpec(spec);
-    options = BackgroundBlurVideoFrameProcessor.resolveOptions(options);
-    const { logger } = options;
+    let processorSpecs: BackgroundFilterSpec = { ...spec };
+    let processorOptions: BackgroundBlurOptions = { ...options };
```
Comment (richhx) [inline src/backgroundblurprocessor/BackgroundBlurVideoFrameProcessor.ts:66]: I would move this into `resolveSpec` and `resolveOptions` and leave the previous code alone.

File: src/backgroundblurprocessor/BackgroundBlurVideoFrameProcessor.ts [hunk 2]
```diff
@@ -112,8 +118,11 @@ export default class BackgroundBlurVideoFrameProcessor extends BackgroundFilterV
    * @returns a boolean promise that will resolve to true if supported and false if not
    */
   static isSupported(spec?: BackgroundFilterSpec, options?: { logger?: Logger }): Promise<boolean> {
-    spec = BackgroundBlurVideoFrameProcessor.resolveSpec(spec);
-    options = BackgroundBlurVideoFrameProcessor.resolveOptions(options);
-    return super.isSupported(spec, options);
+    let processorSpecs: BackgroundFilterSpec = { ...spec };
+    let processorOptions: BackgroundBlurOptions = { ...options };
```
Comment (richhx) [inline src/backgroundblurprocessor/BackgroundBlurVideoFrameProcessor.ts:122]: Same (i.e. I would do the shallow clone in `resolveSpec` and `resolveOptions`)

File: src/backgroundreplacementprocessor/BackgroundReplacementVideoFrameProcessor.ts [hunk 3]
```diff
@@ -60,20 +60,26 @@ export default class BackgroundReplacementVideoFrameProcessor extends Background
     spec?: BackgroundFilterSpec,
     options?: BackgroundReplacementOptions
   ): Promise<BackgroundReplacementProcessor | undefined> {
-    spec = this.resolveSpec(spec);
-    options = this.resolveOptions(options);
-    await this.resolveOptionsAsync(options);
-    const { logger } = options;
+    let processorSpecs: BackgroundFilterSpec = { ...spec };
+    let processorOptions: BackgroundReplacementOptions = { ...options };
```
Comment (richhx) [inline src/backgroundreplacementprocessor/BackgroundReplacementVideoFrameProcessor.ts:64]: Same.

File: src/backgroundreplacementprocessor/BackgroundReplacementVideoFrameProcessor.ts [hunk 4]
```diff
@@ -121,19 +127,22 @@ export default class BackgroundReplacementVideoFrameProcessor extends Background
     spec?: BackgroundFilterSpec,
     options?: BackgroundReplacementOptions
   ): Promise<boolean> {
-    spec = this.resolveSpec(spec);
-    options = this.resolveOptions(options);
-    await this.resolveOptionsAsync(options);
-    const imageBlob = options.imageBlob;
+    let processorSpecs: BackgroundFilterSpec = { ...spec };
+    let processorOptions: BackgroundReplacementOptions = { ...options };
```
Comment (richhx) [inline src/backgroundreplacementprocessor/BackgroundReplacementVideoFrameProcessor.ts:131]: same

File: test/backgroundprocessor/BackgroundBlurProcessor.test.ts [hunk 5]
```diff
@@ -960,4 +961,36 @@ describe('BackgroundBlurProcessor', () => {
       );
     });
   });
+
+  describe('BackgroundBlurVideoFrameProcessor', () => {
+    const spec: BackgroundFilterSpec = {};
+    const options: BackgroundBlurOptions = {
+      blurStrength: BlurStrength.MEDIUM,
```
Comment (richhx) [inline test/backgroundprocessor/BackgroundBlurProcessor.test.ts:968]: Change this to something besides `BlurStrength.MEDIUM`? By default, it'll use the medium so you wouldn't necessarily be able to tell if it was modified.

Comment (hekmatba) [reply test/backgroundprocessor/BackgroundBlurProcessor.test.ts:968]: Good point, I can change that. The resolveOptions would append couple more things so we would be able to tell if it was changed.

File: src/backgroundblurprocessor/BackgroundBlurVideoFrameProcessor.ts (hunk 1 above)
Comment (hekmatba) [reply src/backgroundblurprocessor/BackgroundBlurVideoFrameProcessor.ts]: done

File: test/backgroundprocessor/BackgroundBlurProcessor.test.ts [hunk 6]
```diff
@@ -960,4 +961,36 @@ describe('BackgroundBlurProcessor', () => {
       );
     });
   });
+
+  describe('BackgroundBlurVideoFrameProcessor', () => {
+    const spec: BackgroundFilterSpec = {};
+    const options: BackgroundBlurOptions = {
+      blurStrength: BlurStrength.LOW,
+    };
```
Comment (richhx) [inline test/backgroundprocessor/BackgroundBlurProcessor.test.ts:969]: Scope this to each `it` block instead so the individual tests don't affect each other?

File: test/backgroundreplacementprocessor/BackgroundReplacementProcessor.test.ts [hunk 7]
```diff
@@ -238,4 +239,37 @@ describe('BackgroundReplacementProcessor', () => {
       });
     });
   });
+
+  describe('BackgroundReplacementVideoFrameProcessor', () => {
+    const spec: BackgroundFilterSpec = undefined;
+    const options: BackgroundReplacementOptions = {
+      imageBlob: undefined,
+    };
```
Comment (richhx) [inline test/backgroundreplacementprocessor/BackgroundReplacementProcessor.test.ts:247]: Same.

**Review by richhx — COMMENTED** [review]
Fix the merge conflict?

File: test/backgroundprocessor/BackgroundBlurProcessor.test.ts (hunk 6 above)
Comment (hekmatba) [reply test/backgroundprocessor/BackgroundBlurProcessor.test.ts:969]: Sure. I included them in a `beforeEach` which should re-initialized before each `it`

**Review by ltrung — APPROVED** [review]

**Pull request merged**